pkg/nsm/apis/nsmconnect/nsmconnect.proto

```diff
@@ -84,7 +84,18 @@ message DiscoveryResponse {
     repeated  netmesh.NetworkService network_service = 1;
 }
 
+// ChannelAdvertiseRequest used by NSE to advertise its available channels
+message ChannelAdvertiseRequest {
```

Comment:
Question:

Do we want to allow Pods to advertise some but not all channels from a Network Service?

Reply:
Based on my understanding from slide 20 of your deck, NSE needs to advertise a channel with a network service name where this channel belongs to. On NSM side on receiving this advertisement, 1 - Corresponding Network Channel CRD gets created, 2 - NetworkService CRD needs to be updated with new channel. (if networkservice of the channel does not exist, do we want to create a new networkservice?), 3 - ok is sent back to NSE.
If I have wrong understanding, happy to discuss in greater details.

Reply:
@sbezverk Your understanding if what is written there matches my read of the slide.  I am asking this question not as a criticism of your PR (its right in line with what's written in the slides) but to actually have a conversation about what makes sense :)

Does it make sense to have a Pod that advertises one channel from a Network Service but not the others?

Reply:
@edwarnicke I think it does, because a network service can be backed up by multiple channels so there might be a chance that we have multiple POD offering its channel for the same networkservice. Example `secure-gateway` it can have multiple exit points each offering its channel.  Does it make sense?

Reply:
@sbezverk I think I may have been less than clear.  So lets go to an example.

Say NS Foo has two channels Baz and Bar.

Does it make sense for a Pod to expose Foo.Baz, but not Foo.Bar?

And if so, why not simply make two separate NS's, one for Baz, and one for Bar?

In other words, do we want to require a particular Pod to expose all channels of a Network Service?